File: CPP/biweekly-contest-41/No1.cc
/**
 * Created by Xiaozhong on 2020/12/12.
 * Copyright (c) 2020/12/12 Xiaozhong. All rights reserved.
 */
#include <vector>
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    int countConsistentStrings(string allowed, vector<string> &words) {
        vector<int> mapper(26, 0);
        for (char c : allowed) mapper[c - 'a'] = 1;
        int ans = 0;
        for (const string &word : words) {
            int idx = 0;
            for (; idx < word.length(); ++idx) {
                if (!mapper[word[idx] - 'a']) break;
            }
            if (idx == word.length()) ++ans;
        }
        return ans;
    }
};

File: Java/src/weeklyContest250/No1.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

package weeklyContest250;

import java.util.HashSet;
import java.util.Set;

public class No1 {
    public int canBeTypedWords(String text, String brokenLetters) {
        Set<Character> broken = new HashSet<>();
        for (char c : brokenLetters.toCharArray()) broken.add(c);
        String[] words = text.split(" ");
        int ans = 0;
        for (String word : words) {
            int idx = 0;
            for (; idx < word.length(); ++idx) {
                if (broken.contains(word.charAt(idx))) break;
            }
            if (idx == word.length()) ans++;
        }
        return ans;
    }
}
